src/encoded/static/components/app.js

```diff
@@ -69,277 +87,967 @@ const biosampleTypeList = [
+                const splitHref = this.state.href.split('#');
+                if (splitHref.length >= 2 && splitHref[1][0] !== '!') {
+                    // URL has hash tag, but not the '#!edit' type
+                    const hashTarget = splitHref[1];
+                    const domTarget = document.getElementById(hashTarget);
+                    if (domTarget) {
+                        // DOM has a matching anchor; scroll to it
+                        const elTop = domTarget.getBoundingClientRect().top;
+                        const docTop = document.documentElement.scrollTop || document.body.scrollTop;
+                        const scrollTop = (elTop + docTop) - (window.innerWidth >= 960 ? 75 : 0);
+                        document.documentElement.scrollTop = scrollTop;
+                        document.body.scrollTop = scrollTop;
+                    }
+                }
+            }
+
+            // Avoid popState on load, see: http://stackoverflow.com/q/6421769/199100
+            const register = window.addEventListener.bind(window, 'popstate', this.handlePopState, true);
+            if (window._onload_event_fired) {
+                register();
+            } else {
+                window.addEventListener('load', setTimeout.bind(window, register));
+            }
+        } else {
+            window.onhashchange = this.onHashChange;
+        }
+        window.onbeforeunload = this.handleBeforeUnload;
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        if (nextState) {
+            return !(_.isEqual(nextState, this.state));
+        }
+        return false;
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (!this.state.session || (this.state.session_cookie !== prevState.session_cookie)) {
+            const updateState = {};
+            updateState.session = parseSessionCookie(this.state.session_cookie);
+            if (!updateState.session['auth.userid']) {
+                updateState.session_properties = {};
+            } else if (updateState.session['auth.userid'] !== (this.state.session && this.state.session['auth.userid'])) {
+                this.fetchSessionProperties();
+            }
+            this.setState(updateState);
+        }
+
+        if (this.props) {
+            Object.keys(this.props).forEach((propKey) => {
+                if (this.props[propKey] !== prevProps[propKey]) {
+                    console.log('changed props: %s', propKey);
+                }
+            });
+        }
+        if (this.state) {
+            Object.keys(this.state).forEach((stateKey) => {
+                if (this.state[stateKey] !== prevState[stateKey]) {
+                    console.log('changed state: %s', stateKey);
+                }
+            });
+        }
+
+        const xhr = this.state.contextRequest;
+        if (!xhr || !xhr.xhr_end || xhr.browser_stats) {
+            return;
+        }
+        const browserEnd = 1 * new Date();
```

Comment:
Yeah the `1 *` is one recommended way to convert from string to integer. Strings can’t be subtracted.